src/docker/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM openjdk:12-jdk-alpine
+VOLUME /tmp
+ARG JAR_FILE
+COPY ${JAR_FILE} app.jar
+ENTRYPOINT [ "java", "-jar", "/app.jar" ]
+
+# FROM openjdk:12-jdk-alpine
```

Comment:
I suggest to use multi stage docker build principle.
1. take FROM latest openjdk 17 image
2. related source to docker
4. run gradle build
5. take FROM latest openjre 17
6. copy jar file from the previous step to this one
7. start the application

Here is the example from real project (not ideal but works)

```
FROM amazoncorretto:11 as compile

# Build the jar using gradle wrapper
WORKDIR /app
COPY . /app/
RUN chmod +x gradlew
RUN ./gradlew build

FROM adoptopenjdk/openjdk11:alpine-jre
# Copy the packaged jar app file to a smaller JRE base image
COPY --from=compile "/app/build/libs/app.jar" /etc/app/

EXPOSE 8484

ENTRYPOINT ["java", "-XX:MaxRAMPercentage=80.0", "-jar", "/etc/app/app.jar"]
```